rust/arrow/src/array/equal/structure.rs

```diff
@@ -37,39 +37,20 @@ fn equal_values(
     rhs_start: usize,
     len: usize,
 ) -> bool {
-    let mut temp_lhs: Option<Buffer> = None;
-    let mut temp_rhs: Option<Buffer> = None;
-
     lhs.child_data()
         .iter()
         .zip(rhs.child_data())
         .all(|(lhs_values, rhs_values)| {
             // merge the null data
-            let lhs_merged_nulls = match (lhs_nulls, lhs_values.null_buffer()) {
```

Comment:
@jorgecarleitao I thought you were referring to this part of the code. I removed the one in `mod.rs` because I found that it was computing a duplicate when dealing with just primitive arrays.